internal/store/postgres/job/job_repository.go

```diff
@@ -359,7 +359,7 @@ func (JobRepository) toUpstreams(storeUpstreams []JobWithUpstream) ([]*job.Upstr
 	if err := me.ToErr(); err != nil {
 		return nil, err
 	}
-	return upstreams, nil
+	return job.Upstreams(upstreams).Deduplicate(), nil
```

Comment:
check whether this deduplication is needed? & also the job inspect deduplication is not happening

Reply:
have removed the multiple deduplication and add deduplication for job inspect case